### PR TITLE
provider/aws: Convert protocols to standard format for Security Groups

### PR DESCRIPTION
**Pull request by catsby**
Convert network protocols to their names for keys/state in Security Group and Security Group Rules, fixing issue(s) when using them interchangeably. 

Fixes issues like #5050 where there is a mix and match of using `"6"` and `"tcp"` for the `protocol` value. The name/number reference can be found here:

- https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml

We support a limited set of those, as defined/leveraged from https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/network_acl_entry.go#L81

Right now, if a user specifies `"17"` for the protocol (the code for `upd`), they'll get a diff error when they plan/apply. This is because the AWS API will accept the code or number, but will automatically convert numbers to their matching code, e.g. send `17` and get `udp` back, **except** for `-1` which represents `all`. If you send `-1`, you get `-1` back. 

In this PR we convert all of the numbers referenced in the `network_acl_entry.go` method to their proper name for hashing. Users can use them interchangeably. I've added a statefunc for both Security Groups and Security Group Rules, and double checked our existing tests to make sure there aren't any stray `6` or other non-string protocols used. 

Both `TestAccAWSSecurityGroupRule_Ingress_Protocol` in `security_group_rule_test.go`, and the simple `"tcp"` -> `"6"` in `security_group_test.go` are regression tests; they would fail on master.

### CODE REVIEW

**Comment by stack72** [comment]
This is a great find @catsby :) Will be good to close off this issue

LGTM!

File: builtin/providers/aws/resource_aws_security_group_rule_test.go [hunk 1]
```diff
@@ -545,6 +582,38 @@ resource "aws_security_group_rule" "ingress_1" {
...
+  cidr_block = "10.0.0.0/16"
+
+  tags {
+    Name = "tf-testing"
+  }
+}
+
+resource "aws_security_group" "web" {
+  name        = "terraform_acceptance_test_example"
+  description = "Used in the terraform acceptance tests"
```
Comment (phinze) [inline builtin/providers/aws/resource_aws_security_group_rule_test.go]: I think it's better to omit these to protect us against future collision - let the tags stick around for taxonomy purposes.

**Comment by phinze** [comment]
Looking solid so far!

 * one inline nit in the acctest fixture
 * I believe "icmp" is also handled as a string and normalized, base on the [aws api docs](https://docs.aws.amazon.com/fr_fr/AWSEC2/latest/APIReference/API_AuthorizeSecurityGroupIngress.html)?

**Comment by catsby** [comment]
Discussed `icmp` out of band; it's currently supported but wasn't explicitly mentioned. This branch allows you to use `icmp` or `1` interchangeably .

**Pull request merged**